python_modules/dagster/dagster/_core/storage/migration/utils.py

```diff
@@ -202,8 +202,8 @@ def extract_asset_keys_idx_columns() -> None:
         return
 
     # add timestamp, tags columns to avoid event deserialization
-    op.add_column("asset_keys", db.Column("wipe_timestamp", db.types.TIMESTAMP))
-    op.add_column("asset_keys", db.Column("last_materialization_timestamp", db.types.TIMESTAMP))
+    op.add_column("asset_keys", db.Column("wipe_timestamp", db.types.TIMESTAMP))  # type: ignore
```

Comment:
this one is very mystifying to me - the sqlalchemy.types module resolves in some situations but not in others

Reply:
Errors appear here because `sqlalchemy.types` is not imported in this module, and the `sqlalchemy/__init__.py` doesn't explictly bind `types`-- instead access to `types` is predicated on the side effect of binding the types attr on`sqlalchemy` at the time that `sqlalchemy.types` is actually imported. It's an annoying aspect of python import system.

Normally we could solve this by just making sure we have `import sqlalchemy.types` at the top, but in this case that won't work because we are going through the `db` alias. The solution I've used elsewhere in our codebase is to do:

```
import sqlalchemy.types as db_types
```

Then access the types as `db_types` instead of `db.types`.